psl-core/src/main/java/org/linqs/psl/reasoner/sgd/SGDReasoner.java

```diff
@@ -226,6 +240,48 @@ private float calculateAnnealedLearningRate(int iteration) {
         }
     }
 
+    public SGDExtension getSgdExtension() {
+        return sgdExtension;
+    }
+
+    public float getAdamBeta1() {
+        return adamBeta1;
+    }
+
+    public float getAdamBeta2() {
+        return adamBeta1;
+    }
+
+    public float getAccumulatedGradientMean(int variableIndex) {
+        this.accumulatedGradientMean = ArrayUtils.ensureCapacity(this.accumulatedGradientMean, variableIndex);
```

Comment:
I think we should skip the ensureCapacity() check here in the get's, and instead just check for out of range and return the default value.

The reason for this is super subtle.
I want these calls to be super fast, since they will happen a lot,
and calling ArrayUtils.ensureCapacity() will probably prevent this function from being fully inline.
Chat me up if this doesn't make sense.